test/unit/org/apache/cassandra/cql3/ast/CreateIndexDDL.java

```diff
@@ -141,7 +141,7 @@ else if (((isFrozen(type) && !type.isVector()) || StorageAttachedIndex.SUPPORTED
         public EnumSet<QueryType> supportedQueries(AbstractType<?> type)
         {
             type = type.unwrap();
-            if (IndexTermType.isEqOnlyType(type))
+            if (IndexTermType.isEqOnlyType(type) || type.isCollection() || type.isUDT() || type.isTuple())
```

Comment:
@maedhroz should we push this into `IndexTermType.isEqOnlyType`?  Its also not 100% true as they do support eq for the sub-elements, just not the field as a whole